src/Bicep.Core/Emit/TemplateWriter.cs

```diff
@@ -311,12 +311,17 @@ private void EmitResource(JsonTextWriter memoryWriter, ResourceSymbol resourceSy
             memoryWriter.WriteStartObject();
 
             var typeReference = EmitHelpers.GetTypeReference(resourceSymbol);
-            var body = resourceSymbol.DeclaringResource.Value;
+            SyntaxBase body = resourceSymbol.DeclaringResource.Value;
             if (body is IfConditionSyntax ifCondition)
             {
                 body = ifCondition.Body;
                 emitter.EmitProperty("condition", ifCondition.ConditionExpression);
             }
+            else if (body is ForSyntax @for)
+            {
+                body = @for.Body;
+                emitter.EmitProperty("copy", () => emitter.EmitCopyObject(resourceSymbol.Name, @for, input: null));
+            }
```

Comment:
Can't body be both for & if?

Reply:
Not right now. That's one of the things that is coming right after 0.3. Forgot to mention it in the description.

---
In reply to: [582212603](https://github.com/Azure/bicep/pull/1521#discussion_r582212603) [](ancestors = 582212603)

Reply:
No, never mind. It's in the description (the "no filtering" bit).

---
In reply to: [582231890](https://github.com/Azure/bicep/pull/1521#discussion_r582231890) [](ancestors = 582231890,582212603)